recipes/CommonVoice/ASR/transformer/train_with_whisper.py

```diff
@@ -0,0 +1,322 @@
+        yield tokens
+
+    sb.dataio.dataset.add_dynamic_item(datasets, text_pipeline)
+
+    # 4. Set output:
+    sb.dataio.dataset.set_output_keys(
+        datasets,
+        ["id", "sig", "tokens_list", "tokens_bos", "tokens_eos", "tokens"],
+    )
+
+    return train_data, valid_data, test_data
+
+
+if __name__ == "__main__":
+    # CLI:
+    hparams_file, run_opts, overrides = sb.parse_arguments(sys.argv[1:])
+
+    # If distributed_launch=True then
+    # create ddp_group with the right communication protocol
+    sb.utils.distributed.ddp_init_group(run_opts)
+
+    with open(hparams_file) as fin:
+        hparams = load_hyperpyyaml(fin, overrides)
+
+    # Create experiment directory
+    sb.create_experiment_directory(
+        experiment_directory=hparams["output_folder"],
+        hyperparams_to_save=hparams_file,
+        overrides=overrides,
+    )
+
+    # Dataset prep (parsing Librispeech)
+    from common_voice_prepare import prepare_common_voice  # noqa
+
+    # multi-gpu (ddp) save data preparation
+    run_on_main(
+        prepare_common_voice,
+        kwargs={
+            "data_folder": hparams["data_folder"],
+            "save_folder": hparams["save_folder"],
+            "train_tsv_file": hparams["train_tsv_file"],
+            "dev_tsv_file": hparams["dev_tsv_file"],
+            "test_tsv_file": hparams["test_tsv_file"],
+            "accented_letters": hparams["accented_letters"],
+            "language": hparams["locale"],
+            "skip_prep": hparams["skip_prep"],
+        },
+    )
+    # Defining tokenizer and loading it
+    tokenizer = hparams["whisper"].tokenizer
+    
+    tokenizer.set_prefix_tokens(hparams["language"], "transcribe", False)
+
+    # we need to prepare the tokens for searchers
+    hparams["valid_greedy_searcher"].set_decoder_input_tokens(
+        tokenizer.prefix_tokens
+    )
+    hparams["valid_greedy_searcher"].set_language_token(
+        tokenizer.prefix_tokens[1]
+    )
+
+    hparams["test_beam_searcher"].set_decoder_input_tokens(
+        tokenizer.prefix_tokens
+    )
+    hparams["test_beam_searcher"].set_language_token(tokenizer.prefix_tokens[1])
+
+    # here we create the datasets objects as well as tokenization and encoding
+    train_data, valid_data, test_data  = dataio_prepare(hparams, tokenizer)
+
+    # Trainer initialization
+    asr_brain = ASR(
+        modules=hparams["modules"],
+        hparams=hparams,
+        run_opts=run_opts,
+        checkpointer=hparams["checkpointer"],
+        opt_class=hparams["whisper_opt_class"],
+    )
+
+    # We load the pretrained whisper model
+    if "pretrainer" in hparams.keys():
+        run_on_main(hparams["pretrainer"].collect_files)
+        hparams["pretrainer"].load_collected(asr_brain.device)
+
+    # We dynamicaly add the tokenizer to our brain class.
+    # NB: This tokenizer corresponds to the one used for Whisper.
+    asr_brain.tokenizer = tokenizer
+    if hparams['test_only'] == False:
```

Comment:
Maybe it is better to use `is` instead of `==`

Reply:
fix